src/Microsoft.Diagnostics.Runtime/src/Utilities/Platform/LinuxFunctions.cs

```diff
@@ -231,6 +232,8 @@ internal override unsafe bool GetFileVersion(string dll, out int major, out int
 
                 address += s_versionLength;
 
+                // TODO:  This should be cleaned up to not read byte by byte in the future.  Leaving it here
```

Comment:
Oh that's a `StreamAddressSpace` wrapping a `FileStream`. Just use the `FileStream` with `Seek` and `Read` instead and eliminate the byte-to-byte reads.